ENH: Default implementation for scripted module

Leengit opened the pull request:
Create a new 3D Slicer custom application [from template](https://github.com/KitwareMedical/SlicerCustomAppTemplate), compare its contents to the present repository, and create this pull request to install a first scripted module.  This template-created scaffolding is the starting point for the module that will run the pytorch model.

~~Update: instead of doing this for just one scripted module `Home`, we are doing this for two scripted modules `VPAWVisualized` and `VPAWModel`.~~

Update: The `Home` module is created via copying files from a fresh use of the 3D Slicer custom application.  Then the 3D Slicer Extension Wizard is used to create `VPAWVisualize` and `VPAWModel`.

Also, miscellaneous: Remove trailing whitespace from lines.  Format Python with `black -C`.

Leengit commented:
I don't have the ability to add reviewers for this pull request.  Would one of @ebrahimebrahim, @andinet, @jcfr, @jadh4v add reviewers ... and/or do a review?  Thank you.

ebrahimebrahim commented:
@jcfr Can we add @Leengit to @KitwareMedical/slicer -- this will resolve the issue with permission to add reviewers, etc.

dzenanz commented:
I invited him.

ebrahimebrahim commented on CMakeLists.txt:
```diff
@@ -35,7 +35,7 @@ set(Slicer_ORGANIZATION_DOMAIN "kitware.com")
 
 # Default home and favorite modules
-set(Slicer_DEFAULT_HOME_MODULE "Home")
+set(Slicer_DEFAULT_HOME_MODULE "VPAWVisualize")
```
I guess this is supposed to configure the application such that the VPAWVisualize module is selected upon launch. But that isn't the case when I launch. IDK why

Leengit replied:
This was a side effect of not using the 3D Slicer Extension Wizard.  It is now resolved.

ebrahimebrahim reviewed (commented):
Great idea to have these two modules and good naming.

Some issues I noticed:
- When I launch, there is no module selected by default. I would think VPAWVisualize should be selected?
- When I launch, the UI can be messed up in one of two ways:
  - If, when I last closed the application, I had VPAWVisualize selected and I had the Slicer UI hidden, then on launch I see no module selector. I need to check _View->Toolbars->Module Selection_  or _View->Module Finder_ and then I can select a module.
  - If I unhide the slicer UI using the cog icon, which becomes visible the first time VPAWVisualize is selected, then after restarting the application I end up with the full slicer UI but no cog icon.

jcfr commented:
> > Can we add @Leengit to https://github.com/orgs/KitwareMedical/teams/slicer -- this will resolve the issue with permission to add reviewers, etc.
> I invited him.

@dzenanz Thanks for taking care of this :pray:

jcfr reviewed (changes requested):
The Home module is special in the sense its role is to customize the application and coordinate a workflow between other "regular" module. Associated logic is not intended to be initialized multiple time.

In this PR, two new modules are being added by copying/renaming the existing Home module. This introduce duplicated code and will hinder future maintenance. 

Instead, I suggest to revisit by running the [Extension Wizard](https://slicer.readthedocs.io/en/latest/user_guide/modules/extensionwizard.html) to create two new modules.

ebrahimebrahim commented:
Thank you @jcfr I didn't realize that was the function of the home module. Really glad you chimed in here. Do you know why this custom app doesn't have one to start with? If you look at the main branch right now there just aren't any modules that are introduced by the custom app (there is no `Modules/` subdirectory at all). There's still [talk of the home module](https://github.com/KitwareMedical/vpaw/blob/a4bfca80dab148c661604d7d01413986e2a545dd/CMakeLists.txt#L38) but there isn't a home module. I dug through the commit history and I don't see a previously existing home module that got deleted so I am a bit puzzled.

@Leengit Extension wizard is definitely the way to go and it's very quick and easy to do. We could just redo this PR with the help of the extension wizard (and of course it's easy to redo the black reformatting thing).

Leengit commented:
Thank you @jcfr, @ebrahimebrahim.  Do we want a total of three modules, `Home`, `VPAWVisualize`, and `VPAWModel` -- if so, what would `Home` be implementing?  (Maybe simply being a start-up / landing screen is functionality enough?)  Alternatively, do we want two modules, `Home` and `VPAWModel`, where `Home` implements the functionality that we've been calling `VPAWVisualize`?

jcfr commented:
The home module will allow to style and orchestrate the overall application workflow.

Implementation of the two modules can be done independently of the home module.

ebrahimebrahim commented:
> Thank you @jcfr, @ebrahimebrahim. Do we want a total of three modules, `Home`, `VPAWVisualize`, and `VPAWModel` -- if so, what would `Home` be implementing? (Maybe simply being a start-up / landing screen is functionality enough?) Alternatively, do we want two modules, `Home` and `VPAWModel`, where `Home` implements the functionality that we've been calling `VPAWVisualize`?

To me the first approach, with three modules, sounds better. This way if we later rethink the overall workflow of the application then we can make changes to the overall workflow that are not entangled with a module that has some specific functionality.

Leengit commented:
I've marked this as a draft for now.  I'll push commits to revert back some missteps, then make changes forward from there.  Ultimately, I will squash these commits into a reasonable set.

ebrahimebrahim reviewed (approved):
Looks great. Tested and it works without any of the problems from the previous review.

I'm happy with merge. If you wanted to you could also add a couple of buttons in the Home module that switch us to the two VPAW modules. Or that could be in a separate PR, either way works

Leengit commented:
> If you wanted to you could also add a couple of buttons in the Home module that switch us to the two VPAW modules. Or that could be in a separate PR, either way works

Good idea.  I'd like to get this PR in, so let's do that in another PR.  I'll submit an issue.

Leengit commented:
I have addressed points raised in the above discussions and we are at a good point to merge so that distinct branches can be worked on from here.  @ebrahimebrahim thank you for the review.  @jcfr @andinet I plan to merge this in 24 hours unless I hear objections from either of you.

Leengit commented:
Squash some commits with this recent force push.
